### PR TITLE
Shopping cart

### PR DESCRIPTION
**Pull request by Vitalii-Kazymyrovych**

### CODE REVIEW

File: src/main/java/com/project/bookstore/service/shopping/cart/ShoppingCartServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    @Override
+    public CartItemWithoutBookTitleDto postItem(
+            String username,
+            CreateCartItemRequestDto requestDto) {
+        ShoppingCart shoppingCart = shoppingCartRepository
+                .findByUserEmail(username)
+                .orElseThrow(
+                        () -> new EntityNotFoundException(
+                                "Can't find shopping cart by username: " + username));
```
Comment (koval-kn) [inline src/main/java/com/project/bookstore/service/shopping/cart/ShoppingCartServiceImpl.java]: it should be created if not found

**Review by koval-kn — CHANGES_REQUESTED** [review]

File: src/main/java/com/project/bookstore/model/CartItem.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+@SQLDelete(sql = "UPDATE cart_items SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
+@Table(name = "cart_items")
+public class CartItem {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne(fetch = FetchType.LAZY)
+    private ShoppingCart shoppingCart;
+    @OneToOne
```
Comment (Rommelua) [inline src/main/java/com/project/bookstore/model/CartItem.java]: ManyToOne

File: src/main/java/com/project/bookstore/service/shopping/cart/ShoppingCartServiceImpl.java (hunk 1 above)
Comment (Rommelua) [reply src/main/java/com/project/bookstore/service/shopping/cart/ShoppingCartServiceImpl.java]: not fixed

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/com/project/bookstore/service/shopping/cart/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+    private final CartItemRepository cartItemRepository;
+    private final UserRepository userRepository;
+
+    @Override
+    public ShoppingCartDto findCartByUserName(String username) {
+
```
Comment (MentorMA) [inline src/main/java/com/project/bookstore/service/shopping/cart/ShoppingCartServiceImpl.java:33]:
```suggestion
```

**Review by MentorMA — APPROVED** [review]

**Pull request merged**